backend/probloom/prob/models.py

```diff
@@ -1,17 +1,85 @@
-from django.db import models
+from typing import Any, Dict
+
+from django.db.models import *
+from django.contrib.auth.models import AbstractUser
+import datetime
 
 # Create your models here.
-class User(models.Model):
-    username = models.CharField(max_length=120, default='')
-    email = models.CharField(max_length=120, default='')
-    password = models.CharField(max_length=120, default='')
-    logged_in = models.BooleanField(default=False)
+class User(AbstractUser):
+    pass
 
-    def __str__(self):
-        return self.username
 
-class UserStatistics(models.Model):
-    lastActiveDays = models.IntegerField(default=0)
+class UserProfile(Model):
+    user = OneToOneField(
+        User, on_delete=CASCADE, related_name="profile", primary_key=True
+    )
+    introduction = TextField(default="")
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "user": self.user.pk,
+            "introduction": str(self.introduction),
+        }
+
+
+class UserStatistics(Model):
+    lastActiveDays = IntegerField(default=0)
+    user = OneToOneField(User, on_delete=CASCADE, related_name="statistics", null=True)
 
     def __str__(self):
         return str(self.lastActiveDays)
+
+
+class ProblemSet(Model):
+    title = CharField(max_length=100, default="default title")
+    date = DateTimeField(auto_now_add=True, blank=True)
+    type = BooleanField(default=False)
+    tag = CharField(max_length=100, default="default tag")
+    difficulty = SmallIntegerField(default=0)
+    content = TextField(max_length=1000, default="default content")
+    creator = ForeignKey(
+        UserStatistics, related_name="created_problem", on_delete=CASCADE
+    )
+    recommender = ManyToManyField(
+        UserStatistics, blank=True, related_name="recommended_problem"
+    )
+    solver = ManyToManyField(UserStatistics, blank=True, related_name="solved_problem")
```

Comment:
Did you deleted solved model? I think it is necessary because we need result of solved problem set.

Reply:
When I added the Solved model to my model, a migration error popped up so I did not add the Solved model to my model. So, it was inevitable to remove it from the test code.